### ISSUE TITLE
Tai calculation 

### ISSUE THREAD

**Issue by imcwx** [yi-jiayu/mahjong.go#3]
https://github.com/yi-jiayu/mahjong.go/blob/d6f7ad6c20767337dec7d07a76aa9f65ec6ae904/win.go#L188

Hey jiayu,

my friends tested/played the game and found the 'tai' calculation to be off on mahjong-party. 
So I went into this repo. I am not familiar with go etc, but from reading the win.go.

it calculated "pong pong" or "pinghu" BEFORE flowers.
and the key thing is it "return" which terminates the function.
and that is done before even calculating the flowers. 
so if we have to sequentially do it, flowers should be calculated first cause it doesn't matter which hand you play.

basically, just need to shift the flowers function upwards.  or maybe remove the "returns" in "pinghu"? 

i guess some other todos are other win/score scenarios?
Not sure if 'zhi mo' , 'hua shang'. '13 yao' , 'da shi si', ' 3 dragons' etc are inside.

**Comment by yi-jiayu** [comment]
Hello,

Thanks for your interest in Mahjong Party! 

You're right, the early return in the chou ping hu case was problematic, I've added a test and fixed it in https://github.com/yi-jiayu/mahjong.go/commit/8587a910a46312376a4f6e8947758c9527c47d4e, and deployed the changes.

Yup, there's still a lot of unfinished work left on Mahjong Party such as other scoring scenarios. Thanks for the help! I'll try to spend more time on it since it's been getting some attention recently haha.

**Comment by imcwx** [comment]
Thank you!
Will close based on ping hu issue. 
Other tai calculation and review is for future implementation.
 Thanks jiayu!
